File: frontend/src/types/previewSession.ts
export type PreviewRoleStatus = 'starting' | 'running' | 'stopped' | 'failed';
export type PreviewSessionStatus = 'starting' | 'running' | 'stopped' | 'failed';

export interface PreviewSessionRole {
	id: string;
	worker_id: string;
	worker_name: string;
	role: string;
	port?: number;
	status: PreviewRoleStatus;
	preview_url?: string;
	error_message?: string;
	updated_at: string;
}

export interface PreviewSession {
	id: string;
	status: PreviewSessionStatus;
	error?: string;
	roles: PreviewSessionRole[];
	created_at: string;
	stopped_at?: string;
}

export interface CreatePreviewSessionRolePayload {
	worker_id: string;
	role: string;
}

export interface CreatePreviewSessionPayload {
	roles: CreatePreviewSessionRolePayload[];
}

File: backend/internal/models/preview_session.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type PreviewSessionStatus string

const (
	PreviewSessionStatusStarting PreviewSessionStatus = "starting"
	PreviewSessionStatusRunning  PreviewSessionStatus = "running"
	PreviewSessionStatusStopped  PreviewSessionStatus = "stopped"
	PreviewSessionStatusFailed   PreviewSessionStatus = "failed"
)

type PreviewRoleStatus string

const (
	PreviewRoleStatusStarting PreviewRoleStatus = "starting"
	PreviewRoleStatusRunning  PreviewRoleStatus = "running"
	PreviewRoleStatusStopped  PreviewRoleStatus = "stopped"
	PreviewRoleStatusFailed   PreviewRoleStatus = "failed"
)

type PreviewSession struct {
	ID        uuid.UUID            `db:"id"`
	CreatedBy uuid.UUID            `db:"created_by"`
	Status    PreviewSessionStatus `db:"status"`
	Error     *string              `db:"error"`
	CreatedAt time.Time            `db:"created_at"`
	UpdatedAt time.Time            `db:"updated_at"`
	StoppedAt *time.Time           `db:"stopped_at"`
}

type PreviewSessionRole struct {
	ID           uuid.UUID         `db:"id"`
	SessionID    uuid.UUID         `db:"session_id"`
	WorkerID     uuid.UUID         `db:"worker_id"`
	Role         string            `db:"role"`
	Port         *int              `db:"port"`
	Status       PreviewRoleStatus `db:"status"`
	ErrorMessage *string           `db:"error_message"`
	PreviewURL   *string           `db:"preview_url"`
	LogOutput    string            `db:"log_output"`
	CreatedAt    time.Time         `db:"created_at"`
	UpdatedAt    time.Time         `db:"updated_at"`
}
